crates/iceberg/src/io/storage.rs

```diff
@@ -79,6 +87,16 @@ impl Storage {
     ) -> crate::Result<(Operator, &'a str)> {
         let path = path.as_ref();
         match self {
+            #[cfg(feature = "storage-memory")]
+            Storage::Memory { config } => {
+                let op = config.build(path)?;
+
+                if let Some(stripped) = path.strip_prefix("file:/") {
+                    Ok((op, stripped))
```

Comment:
nit: I don't personally that like stripping `file:/` is necessary. 
If users want to use file-system like naming inside memory-storage, I would just let them 🤷 
Otherwise, you could argue that you should also strip things like `s3:/` as well 😄

Reply:
Oh, it's a typo. We should trim `memory:/` instead.